.github/workflows/Benchmark.yml

```diff
@@ -19,23 +23,29 @@ concurrency:
     cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
 jobs:
     benchmark:
-        if: ${{ github.event.issue.pull_request && contains(github.event.comment.body, '/run-benchmark') }}
-
+        if: |
+            (github.event.issue.pull_request && contains(github.event.comment.body, '/run-benchmark')) ||
+            github.event_name == 'workflow_dispatch'
         runs-on: ubuntu-latest
-
         steps:
             - name: Find PR repo and ref
               uses: actions/github-script@v6
               id: find-ref
               with:
                   script: |
-                      const pr = await github.rest.pulls.get({
-                        owner: context.repo.owner,
-                        repo: context.repo.repo,
-                        pull_number: context.issue.number
-                      })
-                      core.setOutput('repo', pr.data.head.repo.full_name)
-                      core.setOutput('ref', pr.data.head.ref)
+                      if (context.eventName == 'pull_request') {
+                          const pr = await github.rest.pulls.get({
+                            owner: context.repo.owner,
+                            repo: context.repo.repo,
+                            pull_number: context.issue.number
+                          })
+                          core.setOutput('repo', pr.data.head.repo.full_name)
+                          core.setOutput('ref', pr.data.head.ref)
+                      }
+                      if (context.eventName == 'workflow_dispatch') {
+                          core.setOutput('repo', '${{ inputs.repo }}')
+                          core.setOutput('ref', '${{ inputs.ref }}')
+                      }
```

Comment:
The benchmark runs against `TulipaEnergy/TulipaEnergyModel.jl#main`, so we might want to check that situation and raise an error instead of comparing it to itself. What do you think?

Reply:
Hmm, good point. I'll fix that

Reply:
I added an if block that checks for this, and fails explicitly